nuxeo-features/nuxeo-annotation/nuxeo-annotation-core/src/main/java/org/nuxeo/ecm/annotation/AnnotationServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_CREATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DOCUMENT_ID_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_DOC_TYPE;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_FLAGS_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_ID_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_LAST_MODIFIER_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_NAME_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_OPACITY_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_PAGE_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_POSITION_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_SECURITY_PROPERTY;
+import static org.nuxeo.ecm.annotation.AnnotationConstants.ANNOTATION_SUBJECT_PROPERTY;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @since 10.1
+ */
+public class AnnotationServiceImpl extends DefaultComponent implements AnnotationService {
+
+    private static final Log log = LogFactory.getLog(AnnotationServiceImpl.class);
+
+    protected static final String ANNOTATIONS_FROM_DOCUMENT = "SELECT * FROM Annotation WHERE annotation:annotationId LIKE '%s%%' ";
+
+    @Override
+    public String createAnnotation(DocumentModel documentModel, Annotation annotation) {
+        CoreSession session = documentModel.getCoreSession();
+        String id = documentModel.getRepositoryName() + "-" + documentModel.getId() + "-" + annotation.getDocumentId();
+
+        // Create annotation as a placeless document
+        DocumentModel annotationModel = session.createDocumentModel(null, id, ANNOTATION_DOC_TYPE);
+        annotationModel.setPropertyValue(ANNOTATION_ID_PROPERTY, id);
+        setAnnotationProperties(annotationModel, annotation);
+        annotationModel = session.createDocument(annotationModel);
+        session.save();
+        return annotationModel.getId();
+    }
+
+    @Override
+    public Annotation getAnnotation(DocumentModel documentModel, String annotationId) {
+        DocumentModel annotationModel = documentModel.getCoreSession().getDocument(new IdRef(annotationId));
+        return new AnnotationImpl(annotationModel);
+    }
+
+    @Override
+    public List<Annotation> getAnnotations(DocumentModel documentModel) {
```

Comment:
Now that I think of it we'll probably need another API based on the ARender connector notion of `documentId` which contains the `docId` and the `xpath`. Because that's what the connector will need. Same thing for the other methods, which should all take a `xpath` argument.